interleave.el

```diff
@@ -38,28 +39,48 @@
 ;; M-x interleave
 ;;
 ;; To insert a note for a page, type i.
-;; Navigation is the same as in `doc-view-mode'.
+;; Navigation is the same as in `doc-view-mode'/`pdf-view-mode'.
 
 ;;; Code:
 
 (require 'org)
 
 (require 'doc-view)
-;; Redefining `doc-view-kill-proc-and-buffer' as `interleave--doc-view-kill-proc-and-buffer'
+;; Redefining `doc-view-kill-proc-and-buffer' as `interleave--pdf-kill-proc-and-buffer'
 ;; because this function is obsolete in emacs 25.1 onwards.
-(defun interleave--doc-view-kill-proc-and-buffer ()
+(defun interleave--pdf-kill-proc-and-buffer ()
   "Kill the current converter process and buffer."
   (interactive)
-  (doc-view-kill-proc)
-  (when (eq major-mode 'doc-view-mode)
+  (when (derived-mode-p 'doc-view-mode)
+    (doc-view-kill-proc))
+  (when (or (derived-mode-p 'doc-view-mode)
+            (derived-mode-p 'pdf-view-mode))
     (kill-buffer (current-buffer))))
 
-(defvar *interleave--org-buf* nil "The Org Buffer")
+(defvar *interleave--org-buffer* nil "Org notes buffer")
+(defvar *interleave--pdf-buffer* nil "PDF buffer associated with the notes buffer")
 
 (defvar interleave--window-configuration nil
   "Variable to store the window configuration before interleave mode was
 enabled.")
 
+(if (featurep 'pdf-view) ; if `pdf-tools' is installed
+    (progn
+      ;; Function wrapper for the macro `pdf-view-current-page'
+      (defconst interleave--pdf-current-page-fn                 (lambda () (pdf-view-current-page)))
```

Comment:
@rudolfochrist  This was the reason why your earlier approach to use the change page hook in pdf-view failed. But I realized that at the very end after I changed everything. 

If you had wrapped `pdf-view-current-page` in a `lambda` wrapper, your approach would have worked too. But I did not know that then.